### PR TITLE
Break Observer OpenRoad coupling

### PR DESCRIPTION
**Pull request by cdleary**
This makes the OpenRoadObserver not depend on the concrete
implementation of the OpenRoad object itself, so that the dbSta.cc file
can be moved to the non-application target dbSta_lib.

### CODE REVIEW

**Comment by cdleary** [comment]
@maliberty Ready to review

File: include/ord/OpenRoadObserver.hh [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+class OpenRoad;
+
+// Observer interface
+class OpenRoadObserver
+{
+ public:
+  virtual ~OpenRoadObserver()
+  {
+    if (unregister_observer_ != nullptr) {
```
Comment (maliberty) [inline include/ord/OpenRoadObserver.hh:30]:
```suggestion
    if (unregister_observer_) {
```

File: src/upf/src/upf.cpp [hunk 2]
```diff
@@ -41,6 +41,7 @@
 
 #include "db_sta/dbNetwork.hh"
 #include "db_sta/dbSta.hh"
+#include "ord/OpenRoad.hh"
```
Comment (maliberty) [inline src/upf/src/upf.cpp:44]: Why is this needed?  It seems to go against the grain of this change.

**Review by maliberty — CHANGES_REQUESTED** [review]

File: src/upf/src/upf.cpp (hunk 2 above)
Comment (cdleary) [reply src/upf/src/upf.cpp:44]: It was implicitly a necessary header that happened ot be dragged in by `dbSta.hh` before -- since we removed the inclusion from there we have to include what we use in this file now.

**Review by maliberty — APPROVED** [review]

**Comment by maliberty** [comment]
The docker failures look unrelated
